gluecodium/src/main/resources/templates/java/AbstractNativeList.mustache

```diff
@@ -0,0 +1,205 @@
+            index += 1;
+            return result;
+        }
+
+        @Override
+        public boolean hasPrevious() { return index >= 0; }
+
+        @Override
+        public T previous() {
+            if (!hasPrevious()) {
+                throw new NoSuchElementException();
+            }
+
+            final T result = get(index);
+            index -= 1;
+            return result;
+        }
+
+        @Override
+        public int nextIndex() { return index + 1; }
+
+        @Override
+        public int previousIndex() { return index - 1; }
+        @Override
+        public void remove() { throw new UnsupportedOperationException(); }
+        @Override
+        public void set(T e) { throw new UnsupportedOperationException(); }
+        @Override
+        public void add(T e) { throw new UnsupportedOperationException(); }
+    }
+
+    private Integer _size = null;
+
+    protected AbstractNativeList(final long nativeHandle, final Disposer disposer) {
+        super(nativeHandle, disposer);
+    }
+
+    protected abstract int getSize();
+
+    private List<Object> toArrayList() {
+        List<Object> arrayList = new ArrayList<>();
+        iterator().forEachRemaining(arrayList::add);
+        return arrayList;
+    }
+
+    @Override
+    public int size() {
+        if (_size == null) _size = getSize();
+        return _size;
+    }
+
+    @Override
+    public boolean isEmpty() { return size() == 0; }
+
+    @Override
+    public boolean contains(Object o) { return toArrayList().contains(o); }
+
+    @Override
+    public Iterator<T> iterator() { return listIterator(0); }
+
+    @Override
+    public Object[] toArray() { return toArrayList().toArray(); }
+
+    @Override
+    public <T> T[] toArray(T[] a) { return toArrayList().toArray(a); }
+
+    @Override
+    public boolean add(T t) { throw new UnsupportedOperationException(); }
+    @Override
+    public boolean remove(Object o) { throw new UnsupportedOperationException(); }
+
+    @Override
+    public boolean containsAll(Collection<?> c) { return toArrayList().containsAll(c); }
+
+    @Override
+    public boolean addAll(Collection<? extends T> c) { throw new UnsupportedOperationException(); }
+    @Override
+    public boolean addAll(int index, Collection<? extends T> c) { throw new UnsupportedOperationException(); }
+    @Override
+    public boolean removeAll(Collection<?> c) { throw new UnsupportedOperationException(); }
+    @Override
+    public boolean retainAll(Collection<?> c) { throw new UnsupportedOperationException(); }
+    @Override
+    public void clear() { throw new UnsupportedOperationException(); }
+
+    @Override
+    public T set(int index, T element) { throw new UnsupportedOperationException(); }
+    @Override
+    public void add(int index, T element) { throw new UnsupportedOperationException(); }
+    @Override
+    public T remove(int index) { throw new UnsupportedOperationException(); }
+
+    @Override
+    public int indexOf(Object o) {
```

Comment:
Potentially this and operation below might cause huge slow down without caching

Reply:
Yes. These are not in the expected usage scenario though. However, might as well throw an exception here now, change to something else later if the scenario changes.